src/data/serializer/Vector.js

```diff
@@ -575,14 +575,15 @@ Ext.define('GeoExt.data.serializer.Vector', {
          * happened in a previous call.
          *
          * @param {Object} obj The object to get the uid of.
+         * @param {String} geometrType The geometryType for the style.
          * @return {String} The uid of the object.
          * @private
          */
-        getUid: function(obj) {
+        getUid: function(obj, geometryType) {
             if (!Ext.isObject(obj)) {
                 Ext.raise('Cannot get uid of non-object.');
             }
-            var key = this.GX_UID_PROPERTY;
+            var key = this.GX_UID_PROPERTY + '-' + geometryType;
```

Comment:
Maybe you can make the `geometryType` optional making the change backwards compatible? (Disclaimer: I'm unsure if anyone else uses this function or if it's just an internal function, in which case this doesn't matter)

Reply:
It is tagged with `@private` so i hope it`s not used outside, but making it optional is still a good idea.